src/libMediator/Mediator.cpp

```diff
@@ -93,4 +92,26 @@ void Mediator::UpdateTxBlockRand(bool isGenesis)
         randVec = sha2.Finalize();
         copy(randVec.begin(), randVec.end(), m_txBlockRand.begin());
     }
+}
+
+std::string Mediator::GetNodeMode(const Peer& peer)
+{
+    std::lock_guard<mutex> lock(m_mutexDSCommitteeNetworkInfo);
+    if (std::find(m_DSCommitteeNetworkInfo.begin(),
+                  m_DSCommitteeNetworkInfo.end(), peer)
+        != m_DSCommitteeNetworkInfo.end())
+    {
+        if (peer == m_DSCommitteeNetworkInfo[0])
+        {
+            return "DSLD";
+        }
+        else
+        {
+            return "DSBU";
+        }
+    }
+    else
+    {
+        return "SHRD";
```

Comment:
Is there any easier way to identify whether this node is shard node or not. This code looks inefficient (despite the impact shouldn't be big)

Reply:
Not really. But just realized that this function can only tell whether the peer is lost due to winning PoW1. This function is called during ProcessSetShardingInfo in Lookup. The DSCommitteeInfo is obtained earlier than the sharding information, so to be simple can just check whether this peer is happened to be the first element in m_DSCommitteeInfo.